extension.js

```diff
@@ -193,43 +163,49 @@ function setClipboard(text) {
     St.Clipboard.get_default().set_text(St.ClipboardType.PRIMARY, text);
 }
 
-/** creates a log menu items
+/** creates a button for a primary container action
  *
- * @param {Container} container is the target container
- */
-function createLogMenuItem(container) {
-    let i = new ContainerMenuItem(container.name, "logs", () => container.logs());
-    i.insert_child_at_index(createIcon("document-open-symbolic.symbolic", "action-logs"), 1);
-    return i;
-}
-
-/** creates a top menu item
- *
- * @param {Container} container is the target container
- */
-function createTopMenuItem(container) {
-    const i = new ContainerMenuItem(container.name, "top", () => container.watchTop());
-    i.insert_child_at_index(createIcon("view-reveal-symbolic.symbolic", "action-top"), 1);
-    return i;
-}
-
-/** creates a shell menu item
- *
- * @param {Container} container is the target container
- */
-function createShellMenuItem(container) {
-    const i = new ContainerMenuItem(container.name, "sh", () => container.shell());
-    i.insert_child_at_index(new St.Label({style_class: "action-sh", text: ">_"}), 1);
-    return i;
+ * @param {Function} command is the action executed when clicking the button
+ * @param {string} iconName is the icon name
+ * */
+function createActionButton(command, iconName) {
+    const btn = new St.Button({
+        track_hover: true,
+        style_class: "containers-action-button button",
+        x_expand: true,
+        x_align: Clutter.ActorAlign.CENTER,
+    });
+    btn.child = new St.Icon({
+        icon_name: iconName,
+        style_class: "popup-menu-icon",
+    });
+    btn.connect("clicked", () => {
+        command();
```

Comment:
It would be good to close the menu here.

Most of the buttons affect the state, but any changes are only picked up when refreshing containers again (the next time the menu is opened)